test/infrastructure/inmemory/internal/server/certs.go

```diff
@@ -26,13 +26,19 @@ import (
 	"sigs.k8s.io/cluster-api/util/certs"
 )
 
-// newCertAndKey builds a new cert and key signed with the given CA and with the given config.
-func newCertAndKey(caCert *x509.Certificate, caKey *rsa.PrivateKey, config *certs.Config) (*x509.Certificate, *rsa.PrivateKey, error) {
-	key, err := certs.NewPrivateKey()
+var key *rsa.PrivateKey
+
+func init() {
```

Comment:
Generating a private key is slow. Now we only do it once on startup and share it across clusters